x/monitoringp/module.go

```diff
@@ -168,7 +166,14 @@ func (am AppModule) ExportGenesis(ctx sdk.Context, cdc codec.JSONCodec) json.Raw
 func (AppModule) ConsensusVersion() uint64 { return 2 }
 
 // BeginBlock executes all ABCI BeginBlock logic respective to the capability module.
-func (am AppModule) BeginBlock(_ sdk.Context, _ abci.RequestBeginBlock) {}
+func (am AppModule) BeginBlock(ctx sdk.Context, bb abci.RequestBeginBlock) {
+	// reports signatures for the block
+	am.keeper.ReportBlockSignatures(ctx, bb.LastCommitInfo, bb.Header.Height)
+
+	// check and transmit signatures
+	// TODO(): investigate proper error handling
+	_ = am.keeper.TransmitSignatures(ctx, bb.Header.Height)
```

Comment:
```suggestion
	err := am.keeper.TransmitSignatures(ctx, bb.Header.Height)
	if err != nil {
		ctx.Logger().Error(fmt.Sprintf("error to transmit the validator signatures %s", err.Error()))
	}
```

Reply:
That's a good suggestion
Also, FYI we have created a task to investigate the best way to handle error in begin block